core/src/main/java/io/grpc/internal/ServerCallImpl.java

```diff
@@ -263,10 +265,13 @@ public SecurityLevel getSecurityLevel() {
    * run until completion, but silently ignore interactions with the {@link ServerStream} from now
    * on.
    */
-  private void internalClose(Status internalError) {
-    log.log(Level.WARNING, "Cancelling the stream with status {0}", new Object[] {internalError});
-    stream.cancel(internalError);
-    serverCallTracer.reportCallEnded(internalError.isOk()); // error so always false
+  private void internalClose(Throwable internalError) {
+    log.log(Level.WARNING, "Cancelling the stream because of internal error}", internalError);
+    Status status = (internalError instanceof StatusRuntimeException)
+        ? ((StatusRuntimeException) internalError).getStatus()
+        : Status.INTERNAL.withCause(internalError);
```

Comment:
Always include a description in the status. Even a random string is helpful as it identifies what generated the status.